word_search.py

```diff
@@ -0,0 +1,41 @@
+TC:Not sure
```

Comment:
What if the whole board is same as the word.  So it must be O(mxn x mxn^L) where L is length of word. It can be more tight still. Its better to be at Exponential.